src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,85 @@
-export const Pagination = () => {};
+import { FC } from 'react';
+import classNames from 'classnames';
+import { getNumbers } from '../../utils';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const numberOfPages = Math.ceil(total / perPage);
+  const pages = [];
+
+  for (let page = 1; page <= numberOfPages; page += 1) {
+    pages.push(
+      <li className={classNames('page-item', { active: page === currentPage })}>
+        <a
+          data-cy="pageLink"
+          className="page-link"
+          href={`#${page}`}
+          onClick={() => onPageChange(page)}
+        >
+          {page}
+        </a>
+      </li>,
+    );
+  }
+
+  const items = getNumbers(1, 42)
+    .map(n => `Item ${n}`);
+
+  const startItem = (currentPage - 1) * perPage;
+  const endItem = startItem + perPage;
+
+  const preparedItems = items.slice(startItem, endItem);
```

Comment:
Let's move it to `app` component.
The pagination component should render only the pagination bar.